### DISCUSSION TITLE
Downloadable binary does not include UHDR file format

### DISCUSSION THREAD

**Discussion by pmerle** [ImageMagick/ImageMagick#7339] [Development]
### ImageMagick version

7.1.1-32

### Operating system, version and so on

Windows

### Description

Although the description says "High-dynamic-range imaging enabled", the build does not include the new file format "uhdr:" which was added 6 weeks ago. Took me some time to find out.

Would it be possible to include the format into the build?

**Comment by fmw42** [comment]
Did you include the libultrahdr library in your build?  See https://github.com/google/libultrahdr

From the formats page:

"[UHDR](https://developer.android.com/media/platform/hdr-image-format)	RW	Ultra HDR	Requires the [libultrahdr](https://github.com/google/libultrahdr.git) delegate library."

So it seems to be an optional format, not built automatically.

**Comment by pmerle** [comment]
No, I am referring to the pre-built binaries made by the developers for the github page and for imagemagick.org.

**Reply by fmw42** [reply]
I suspect that the developers do not want to automatically include that. But I will defer to them to reply.

**Comment by dlemstra** [comment]
There is no support for the UHDR format in the Windows build of Imagemagick yet. This might be added in the future.

**Comment by pmerle** [comment]
Is there on any of the Linux builds? Why the difference?

**Comment by dlemstra** [comment]
We maintain the Windows build and the dependencies so that means we would need to include the `libultrahdr` library in our build. For other platforms the build checks if it is available and then uses it.

**Comment by pmerle** [comment]
Would be great if you include it!

This format gets popular these days because Android 14 just starts to use it for HDR images. The OS updates rolled out now save HDR pictures in Ultra HDR, while this was a proprietary format on Android smartphones before. And unfortunately there is no editing software that supports it yet, like to crop or resize.
